### PR TITLE
[WIP] Add example that illustrates how to customize waypoints

### LINKED ISSUE #21: Show how to customize waypoints
An example of how to customize waypoints would be useful. I'll try to start an example that illustrates how to distinctly style waypoints, such as for using different icons for pickup and dropoff points.

cc @1ec5

### PR DESCRIPTION
**Pull request by captainbarbosa**
Closes #21 

Adds a new example that explains how to customize the style of individual waypoints.

### CODE REVIEW

File: Navigation-Examples/Examples/Styled-Waypoints.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        
+        Directions.shared.calculate(options) { (waypoints, routes, error) in
+            guard let route = routes?.first, error == nil else {
+                print(error!.localizedDescription)
+                return
+            }
+            
+            let navigationController = NavigationViewController(for: route)
+            let navigationMapView = NavigationMapView()
+            navigationMapView.navigationMapDelegate = self
```
Comment (captainbarbosa) [inline Navigation-Examples/Examples/Styled-Waypoints.swift]: I'm not sure if I'm doing this right, because none of the delegate methods are being called as it stands.

File: Navigation-Examples/Examples/Styled-Waypoints.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    func navigationMapView(_ mapView: NavigationMapView, waypointStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer? {
+        let waypointStyleLayer = MGLCircleStyleLayer(identifier: identifier, source: source)
+        
+        // TODO: Style waypoints here based on "title" attribute
+        waypointStyleLayer.circleColor = MGLStyleValue(rawValue: UIColor.red)
+        return waypointStyleLayer
+    }
+    
+    func navigationMapView(_ mapView: NavigationMapView, waypointSymbolStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer? {
+        return nil
```
Comment (bsudekum) [inline Navigation-Examples/Examples/Styled-Waypoints.swift]: Does the symbol still appear when you do this? If you goal is hide the symbol, I'd suggest building the layer then calling `layer.isVisible = false` on it.

Comment (captainbarbosa) [reply Navigation-Examples/Examples/Styled-Waypoints.swift]: I'm not sure, https://github.com/mapbox/navigation-ios-examples/pull/22#discussion_r187728306 is blocking me from getting these delegate methods to be called 🤔 am I doing it incorrectly?

File: Navigation-Examples/Examples/Styled-Waypoints.swift (hunk 1 above)
Comment (bsudekum) [reply Navigation-Examples/Examples/Styled-Waypoints.swift]: It's a little hidden. try `navigationController.mapView.navigationMapDelegate = self`.

File: Navigation-Examples/Examples/Styled-Waypoints.swift (hunk 2 above)
Comment (captainbarbosa) [reply Navigation-Examples/Examples/Styled-Waypoints.swift]: Ok, looks like `navigationController.mapView.navigationMapDelegate = self` did the trick.

**Comment by captainbarbosa** [comment]
As of now, the style layer is not being added to the map. I thought that `navigationMapView(_:waypointStyleLayerWithIdentifier:source:)` took care of adding the style layer already under the hood:

File: Navigation-Examples/Examples/Styled-Waypoints.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        
+        Directions.shared.calculate(options) { (waypoints, routes, error) in
+            guard let route = routes?.first, error == nil else {
+                print(error!.localizedDescription)
+                return
+            }
+            
+            let navigationController = NavigationViewController(for: route)
+
+            navigationController.mapView?.navigationMapDelegate = self
```
Comment (1ec5) [inline Navigation-Examples/Examples/Styled-Waypoints.swift]: NavigationViewController wants to be its map view’s delegate, but this overrides the delegate to be StyledWaypointsViewController. This is a bit of a trap in the SDK. (mapbox/mapbox-navigation-ios#685 would turn the trap into the preferred way of doing things.)

Instead, set `NavigationViewController.delegate` to `self` and implement `NavigationViewControllerDelegate` methods instead of `NavigationMapViewDelegate` methods.

File: Navigation-Examples/Examples/Styled-Waypoints.swift [hunk 4]
```diff
@@ -22,8 +22,7 @@ class StyledWaypointsViewController: UIViewController, NavigationMapViewDelegate
             }
             
             let navigationController = NavigationViewController(for: route)
-
-            navigationController.mapView?.navigationMapDelegate = self
+            navigationController.delegate = self
```
Comment (captainbarbosa) [inline Navigation-Examples/Examples/Styled-Waypoints.swift]: @1ec5 I tried your suggestion here, but I'm still seeing no waypoint style layer 😢

**Comment by captainbarbosa** [comment]
@1ec5 I updated to v0.17.0 to test if this would work. It was kind of successful but there is some unexpected behavior I'm seeing:

1. Setting style properties in [`navigationMapView(_:waypointSymbolStyleLayerWithIdentifier:source:)`](https://www.mapbox.com/mapbox-navigation-ios/navigation/0.17.0/Classes/NavigationViewController.html#/c:@M@MapboxNavigation@objc(pl)MBNavigationMapViewDelegate(im)navigationMapView:waypointSymbolStyleLayerWithIdentifier:source:) doesn't seem to have any effect on the waypoint symbol style layer.

2. Despite using multiple waypoints, only the first one is drawn on the display.

A screenshot of both issues is below:

File: Navigation-Examples/Examples/Basic.swift [hunk 5]
```diff
@@ -8,26 +8,6 @@ class BasicViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        let origin = CLLocationCoordinate2DMake(37.77440680146262, -122.43539772352648)
```
Comment (bsudekum) [inline Navigation-Examples/Examples/Basic.swift]: Why was this removed?

Comment (captainbarbosa) [reply Navigation-Examples/Examples/Basic.swift]: @bsudekum I took this out to get v0.17.0 up and running on this branch quickly (mentioned at the end of https://github.com/mapbox/navigation-ios-examples/pull/22#issuecomment-389004114) - I will rebase this branch against master now that https://github.com/mapbox/navigation-ios-examples/pull/20 has merged to revert this.

**Comment by captainbarbosa** [comment]
Prob won't be able to get back to this anytime soon 🌆
